### PR TITLE
Add support for variable names to the NameSectionReader.

### PR DESCRIPTION
**Pull request by bmeurer**
@yurydelendik this is a proof of concept, looking for advice how to pursue this. Especially the fixing of bad names in `getNameResolver`, is that still needed?

### CODE REVIEW

File: src/WasmDis.ts [hunk 1]
```diff
@@ -1096,20 +1099,29 @@ class NameSectionNameResolver extends DefaultNameResolver {
       return `$${UNKNOWN_FUNCTION_PREFIX}${index}`;
     return isRef ? `$${name}` : `$${name} (;${index};)`;
   }
+
+  getVariableName(funcIndex: number, index: number, isRef: boolean): string {
+    const name = this._localNames[funcIndex] && this._localNames[funcIndex][index];
+    if (!name)
+      return `$${UNKNOWN_FUNCTION_PREFIX}${index}`;
```
Comment (yurydelendik) [inline src/WasmDis.ts]: let's use DefaultNameResolver's method if name is not specified: `super.getVariableName(funcIndex, index, isRef)` ?

Comment (bmeurer) [reply src/WasmDis.ts]: Done.

**Comment by bmeurer** [comment]
Cool, added the `super` call. I'll add some tests as well.

**Comment by bmeurer** [comment]
Ok, there you go, ready to land with tests.

**Comment by yurydelendik** [comment]
Thank you for the patch

**Pull request merged**